### DISCUSSION TITLE
Installing the complete Accounting module

### DISCUSSION THREAD

**Discussion by ubidev** [Tecnativa/doodba#613] [Doodba - The template] [answered]
Hi!
I'm trying to install the full accounting module via Doodba.
I tried via addons.yaml but it does not install. 
I tried via the addon script with same results but other modules are installed and show up in the GUI.

`addons -w account_accountant -w account -W pad -W pad_project -W note_pad -i -e -n -c init`

Now, I was kindly told in another discussion here that if using addons.yaml, I have to install it myself. How do I do that since that the apps web page for installing the modules show an UPGRADE button that sends me to a paid subscription page instead of an INSTALL button?
![image](https://github.com/Tecnativa/doodba/assets/14218601/4966899c-88b4-4ab8-863d-030804433973)

And why isn't the accounting module installed like the others when using the addons script?

Thanks a zilion for your time

**Comment by marcelsavegnago** [comment]
Did you add the odoo/enterprise repository to addons.yaml?

**Reply by ubidev** [reply]
Unfortunately no.
How would I proceed as I have found no references to that effect in the Readme of Doodba and my spanish is not strong enough at this time to understand properly the 2 videos I found referencing Doodba deployment ;-)

Thanks for your help

**Comment by pedrobaeza** [comment]
With a copier template populated, you can do `invoke install -n -o` for installing all the existing enterprise and core modules. See the help of the command:

```
Usage: inv[oke] [--core-opts] install [--options] [other tasks here ...]

Docstring:
  Install Odoo addons

  By default, installs addon from directory being worked on,
  unless other options are specified.

Options:
  -c STRING, --cur-file=STRING   Path to the current file. Addon name will be obtained from there to install.
  -e, --extra                    Install all extra addons. Default: False
  -m STRING, --modules=STRING    Comma-separated list of modules to install.
  -n, --enterprise               Install all enterprise addons. Default: False
  -o, --core                     Install all core addons. Default: False
  -p, --private                  Install all private addons. Default: False
```

**Comment by ubidev** [comment]
I like this suggestion for the fact that it can be run directly in wsl outside the container. Unfortunately, problems arise :(

First, invoke install cannot be passed an exception list as with the -W parameter of the addons script. Currently I need to exclude de "pad" related modules as trying to install them generates a breaking error:
2024-06-05 15:52:13,666 20 ERROR devel odoo.modules.registry: Failed to load registry
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 169, in _new_conn
    conn = connection.create_connection(
  File "/usr/local/lib/python3.8/site-packages/urllib3/util/connection.py", line 73, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/local/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -3] Temporary failure in name resolution

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 353, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 181, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7fc2cedf7490>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/usr/local/lib/python3.8/site-packages/urllib3/util/retry.py", line 574, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='pad.odoo.com', port=443): Max retries exceeded with url: /api/1/createPad (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fc2cedf7490>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/auto/addons/pad/models/pad.py", line 58, in pad_generate_url
    myPad.createPad(path)
  File "/opt/odoo/auto/addons/pad/py_etherpad/__init__.py", line 209, in createPad
    return self.call("createPad", params)
  File "/opt/odoo/auto/addons/pad/py_etherpad/__init__.py", line 40, in call
    r = requests.post(url, data=params, timeout=self.TIMEOUT)
  File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='pad.odoo.com', port=443): Max retries exceeded with url: /api/1/createPad (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fc2cedf7490>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/modules/registry.py", line 87, in new
    odoo.modules.load_modules(registry, force_demo, status, update_module)
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 474, in load_modules
    processed_modules += load_marked_modules(cr, graph,
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 363, in load_marked_modules
    loaded, processed = load_module_graph(
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 199, in load_module_graph
    registry.init_models(cr, model_names, {'module': package.name}, new_install)
  File "/opt/odoo/custom/src/odoo/odoo/modules/registry.py", line 429, in init_models
    model.init()
  File "/opt/odoo/auto/addons/project_timesheet_holidays/models/res_company.py", line 39, in init
    task = company.env['project.task'].create({
  File "<decorator-gen-305>", line 2, in create
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 395, in _model_create_single
    return create(self, arg)
  File "/opt/odoo/auto/addons/pad_project/models/project.py", line 33, in create
    return super(ProjectTask, self).create(vals)
  File "<decorator-gen-269>", line 2, in create
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 417, in _model_create_multi
    return create(self, [arg])
  File "/opt/odoo/auto/addons/project/models/project.py", line 1777, in create
    tasks = super(Task, self).create(vals_list)
  File "<decorator-gen-125>", line 2, in create
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 418, in _model_create_multi
    return create(self, arg)
  File "/opt/odoo/auto/addons/mail/models/mail_thread.py", line 265, in create
    threads = super(MailThread, self).create(vals_list)
  File "<decorator-gen-200>", line 2, in create
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 398, in _model_create_single
    return self.browse().concat(*(create(self, vals) for vals in arg))
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 398, in <genexpr>
    return self.browse().concat(*(create(self, vals) for vals in arg))
  File "/opt/odoo/auto/addons/pad/models/pad.py", line 133, in create
    pad_info = self.with_context(**ctx).pad_generate_url()
  File "/opt/odoo/auto/addons/pad/models/pad.py", line 60, in pad_generate_url
    raise UserError(_("Pad creation failed, either there is a problem with your pad server URL or with your connection."))
odoo.exceptions.UserError: Pad creation failed, either there is a problem with your pad server URL or with your connection.
2024-06-05 15:52:13,666 20 CRITICAL devel odoo.service.server: Failed to initialize database `devel`.
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 169, in _new_conn
    conn = connection.create_connection(
  File "/usr/local/lib/python3.8/site-packages/urllib3/util/connection.py", line 73, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/local/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -3] Temporary failure in name resolution

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 353, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 181, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7fc2cedf7490>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/usr/local/lib/python3.8/site-packages/urllib3/util/retry.py", line 574, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='pad.odoo.com', port=443): Max retries exceeded with url: /api/1/createPad (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fc2cedf7490>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/auto/addons/pad/models/pad.py", line 58, in pad_generate_url
    myPad.createPad(path)
  File "/opt/odoo/auto/addons/pad/py_etherpad/__init__.py", line 209, in createPad
    return self.call("createPad", params)
  File "/opt/odoo/auto/addons/pad/py_etherpad/__init__.py", line 40, in call
    r = requests.post(url, data=params, timeout=self.TIMEOUT)
  File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='pad.odoo.com', port=443): Max retries exceeded with url: /api/1/createPad (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fc2cedf7490>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/service/server.py", line 1260, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "/opt/odoo/custom/src/odoo/odoo/modules/registry.py", line 87, in new
    odoo.modules.load_modules(registry, force_demo, status, update_module)
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 474, in load_modules
    processed_modules += load_marked_modules(cr, graph,
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 363, in load_marked_modules
    loaded, processed = load_module_graph(
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 199, in load_module_graph
    registry.init_models(cr, model_names, {'module': package.name}, new_install)
  File "/opt/odoo/custom/src/odoo/odoo/modules/registry.py", line 429, in init_models
    model.init()
  File "/opt/odoo/auto/addons/project_timesheet_holidays/models/res_company.py", line 39, in init
    task = company.env['project.task'].create({
  File "<decorator-gen-305>", line 2, in create
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 395, in _model_create_single
    return create(self, arg)
  File "/opt/odoo/auto/addons/pad_project/models/project.py", line 33, in create
    return super(ProjectTask, self).create(vals)
  File "<decorator-gen-269>", line 2, in create
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 417, in _model_create_multi
    return create(self, [arg])
  File "/opt/odoo/auto/addons/project/models/project.py", line 1777, in create
    tasks = super(Task, self).create(vals_list)
  File "<decorator-gen-125>", line 2, in create
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 418, in _model_create_multi
    return create(self, arg)
  File "/opt/odoo/auto/addons/mail/models/mail_thread.py", line 265, in create
    threads = super(MailThread, self).create(vals_list)
  File "<decorator-gen-200>", line 2, in create
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 398, in _model_create_single
    return self.browse().concat(*(create(self, vals) for vals in arg))
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 398, in <genexpr>
    return self.browse().concat(*(create(self, vals) for vals in arg))
  File "/opt/odoo/auto/addons/pad/models/pad.py", line 133, in create
    pad_info = self.with_context(**ctx).pad_generate_url()
  File "/opt/odoo/auto/addons/pad/models/pad.py", line 60, in pad_generate_url
    raise UserError(_("Pad creation failed, either there is a problem with your pad server URL or with your connection."))
odoo.exceptions.UserError: Pad creation failed, either there is a problem with your pad server URL or with your connection.
2024-06-05 15:52:13,667 20 INFO devel odoo.service.server: Initiating shutdown
2024-06-05 15:52:13,667 20 INFO devel odoo.service.server: Hit CTRL-C again or send a second signal to force the shutdown.
2024-06-05 15:52:13,667 20 INFO devel odoo.sql_db: ConnectionPool(used=0/count=0/max=64): Closed 2 connections
Traceback (most recent call last):
  File "/usr/local/bin/addons", line 144, in <module>
    check_call(command)
  File "/usr/local/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['odoo', '--stop-after-init', '--init', 'account,account_check_printing,account_debit_note,account_edi,account_edi_facturx,account_edi_proxy_client,account_edi_ubl,account_edi_ubl_bis3,account_edi_ubl_cii,account_fleet,account_lock,account_payment,account_qr_code_sepa,account_sale_timesheet,account_tax_python,account_test,analytic,association,attachment_indexation,auth_ldap,auth_oauth,auth_password_policy,auth_password_policy_portal,auth_password_policy_signup,auth_signup,auth_totp,auth_totp_mail,auth_totp_mail_enforce,auth_totp_portal,barcodes,barcodes_gs1_nomenclature,base_address_city,base_address_extended,base_automation,base_geolocalize,base_iban,base_import,base_import_module,base_setup,base_sparse_field,base_vat,board,bus,calendar,calendar_sms,contacts,coupon,crm,crm_iap_enrich,crm_iap_mine,crm_livechat,crm_mail_plugin,crm_sms,delivery,delivery_mondialrelay,digest,event,event_booth,event_booth_sale,event_crm,event_crm_sale,event_sale,event_sms,fetchmail,fetchmail_gmail,fetchmail_outlook,fleet,gamification,gamification_sale_crm,gift_card,google_account,google_calendar,google_drive,google_gmail,google_recaptcha,google_spreadsheet,hr,hr_attendance,hr_contract,hr_expense,hr_fleet,hr_gamification,hr_holidays,hr_holidays_attendance,hr_maintenance,hr_org_chart,hr_presence,hr_recruitment,hr_recruitment_survey,hr_skills,hr_skills_slides,hr_skills_survey,hr_timesheet,hr_timesheet_attendance,hr_work_entry,hr_work_entry_contract,hr_work_entry_holidays,http_routing,hw_drivers,hw_escpos,hw_l10n_eg_eta,hw_posbox_homepage,iap,iap_crm,iap_mail,im_livechat,im_livechat_mail_bot,l10n_account_edi_ubl_cii_tests,l10n_ae,l10n_ae_pos,l10n_ar,l10n_ar_website_sale,l10n_at,l10n_au,l10n_be,l10n_be_edi,l10n_bg,l10n_bo,l10n_br,l10n_ca,l10n_ch,l10n_cl,l10n_cn,l10n_cn_city,l10n_co,l10n_co_pos,l10n_cr,l10n_cz,l10n_de,l10n_de_purchase,l10n_de_repair,l10n_de_sale,l10n_de_skr03,l10n_de_skr04,l10n_de_stock,l10n_dk,l10n_do,l10n_dz,l10n_ec,l10n_eg,l10n_eg_edi_eta,l10n_es,l10n_es_edi_sii,l10n_et,l10n_eu_oss,l10n_fi,l10n_fi_sale,l10n_fr,l10n_fr_facturx_chorus_pro,l10n_fr_fec,l10n_fr_pos_cert,l10n_gcc_invoice,l10n_gcc_invoice_stock_account,l10n_gcc_pos,l10n_generic_coa,l10n_gr,l10n_gt,l10n_hk,l10n_hn,l10n_hr,l10n_hu,l10n_id,l10n_id_efaktur,l10n_ie,l10n_il,l10n_in,l10n_in_edi,l10n_in_edi_ewaybill,l10n_in_pos,l10n_in_purchase,l10n_in_purchase_stock,l10n_in_sale,l10n_in_sale_stock,l10n_in_stock,l10n_in_upi,l10n_it,l10n_it_edi,l10n_it_edi_sdicoop,l10n_it_edi_website_sale,l10n_it_stock_ddt,l10n_jp,l10n_ke,l10n_ke_edi_tremol,l10n_latam_base,l10n_latam_invoice_document,l10n_lt,l10n_lu,l10n_lu_peppol_id,l10n_ma,l10n_mn,l10n_multilang,l10n_mx,l10n_my,l10n_nl,l10n_nl_edi,l10n_no,l10n_no_edi,l10n_nz,l10n_pa,l10n_pe,l10n_ph,l10n_pk,l10n_pl,l10n_pt,l10n_ro,l10n_rs,l10n_sa,l10n_sa_edi,l10n_sa_edi_pos,l10n_sa_invoice,l10n_sa_pos,l10n_se,l10n_sg,l10n_si,l10n_sk,l10n_syscohada,l10n_th,l10n_tr,l10n_tw,l10n_ua,l10n_uk,l10n_us,l10n_uy,l10n_ve,l10n_vn,l10n_za,link_tracker,lunch,mail,mail_bot,mail_bot_hr,mail_group,mail_plugin,maintenance,mass_mailing,mass_mailing_crm,mass_mailing_crm_sms,mass_mailing_event,mass_mailing_event_sms,mass_mailing_event_track,mass_mailing_event_track_sms,mass_mailing_sale,mass_mailing_sale_sms,mass_mailing_slides,mass_mailing_sms,membership,microsoft_account,microsoft_calendar,microsoft_outlook,mrp,mrp_account,mrp_landed_costs,mrp_product_expiry,mrp_repair,mrp_subcontracting,mrp_subcontracting_account,mrp_subcontracting_dropshipping,mrp_subcontracting_purchase,note,note_pad,pad,pad_project,partner_autocomplete,payment,payment_adyen,payment_alipay,payment_authorize,payment_buckaroo,payment_mollie,payment_ogone,payment_paypal,payment_payulatam,payment_payumoney,payment_sips,payment_stripe,payment_test,payment_transfer,phone_validation,point_of_sale,portal,portal_rating,pos_adyen,pos_cache,pos_coupon,pos_discount,pos_epson_printer,pos_epson_printer_restaurant,pos_gift_card,pos_hr,pos_mercury,pos_mrp,pos_restaurant,pos_restaurant_adyen,pos_sale,pos_sale_gift_card,pos_sale_margin,pos_sale_product_configurator,pos_six,product,product_email_template,product_expiry,product_images,product_margin,product_matrix,project,project_account,project_hr_expense,project_mail_plugin,project_mrp,project_purchase,project_timesheet_holidays,purchase,purchase_mrp,purchase_product_matrix,purchase_requisition,purchase_requisition_stock,purchase_requisition_stock_dropshipping,purchase_stock,rating,repair,resource,sale,sale_coupon,sale_coupon_delivery,sale_crm,sale_expense,sale_gift_card,sale_management,sale_margin,sale_mrp,sale_mrp_margin,sale_product_configurator,sale_product_matrix,sale_project,sale_project_account,sale_purchase,sale_purchase_stock,sale_quotation_builder,sale_sms,sale_stock,sale_stock_margin,sale_timesheet,sale_timesheet_margin,sales_team,sms,snailmail,snailmail_account,social_media,stock,stock_account,stock_dropshipping,stock_landed_costs,stock_picking_batch,stock_sms,survey,test_base_automation,test_discuss_full,test_event_full,test_mail,test_mail_full,test_mass_mailing,test_website,test_website_modules,test_website_slides_full,test_xlsx_export,theme_default,transifex,uom,utm,web,web_editor,web_kanban_gauge,web_tour,web_unsplash,website,website_blog,website_crm,website_crm_iap_reveal,website_crm_livechat,website_crm_partner_assign,website_crm_sms,website_customer,website_event,website_event_booth,website_event_booth_exhibitor,website_event_booth_sale,website_event_booth_sale_exhibitor,website_event_crm,website_event_crm_questions,website_event_exhibitor,website_event_meet,website_event_meet_quiz,website_event_questions,website_event_sale,website_event_track,website_event_track_live,website_event_track_live_quiz,website_event_track_quiz,website_form_project,website_forum,website_google_map,website_hr_recruitment,website_jitsi,website_links,website_livechat,website_mail,website_mail_group,website_mass_mailing,website_membership,website_partner,website_payment,website_profile,website_sale,website_sale_comparison,website_sale_comparison_wishlist,website_sale_coupon,website_sale_coupon_delivery,website_sale_delivery,website_sale_delivery_giftcard,website_sale_delivery_mondialrelay,website_sale_digital,website_sale_gift_card,website_sale_product_configurator,website_sale_slides,website_sale_stock,website_sale_stock_product_configurator,website_sale_stock_wishlist,website_sale_wishlist,website_slides,website_slides_forum,website_slides_survey,website_sms,website_twitter']' returned non-zero exit status 255.


This error can be avoided by running:

`addons -c -e -i -n -W pad -W pad_project -W note_pad init`

Anyways, thanks to you guys, my current updated understanding is whatever command I use to install enterprise modules (addons script, invoke install, odoo-filter-addons), I need to modify the default repos.yaml for that to work.

Am I on the right track? If yes, what modification do I need in the file and other related changes maybe?

Thanks alot again. I'm sure this precious info will be useful to future Doodba noobs like me. 

Once my understanding is clear, I'll evaluate the possibility of creating a YT video to that effect :-)

**Reply by pedrobaeza** [reply]
With an empty repos.yaml, you will only get as available addons the Odoo core (community) ones + the custom ones you have dropped in the `private` folder. For having enterprise (as it's not open source), you should add the definition on how to get that modules. For that, you can configure a deploy token of one GitHub login authorized to see odoo/enterprise, a reference to a fork, etc, and use it with the syntax of that file, coming from git-aggregator project.

**Reply by ubidev** [reply]
Nice as my customer has enterprise access. So if I understand correctly:

1. I need my client to fork odoo/enterprise in his github account
2. Provide me with a deploy token
3. I add the fork url to repos.yaml
4. I put the deploy token somewhere specific

Right? If so, where do I put the token and how do I add the fork url in the following example?

`
./odoo:
  defaults:
    # Shallow repositories ($DEPTH_DEFAULT=1) are faster & thinner
    # You may need a bigger depth when merging PRs (use $DEPTH_MERGE
    # for a sane value of 100 commits)
    depth: $DEPTH_DEFAULT
  remotes:
    ocb: https://github.com/OCA/OCB.git
    odoo: https://github.com/odoo/odoo.git
    openupgrade: https://github.com/OCA/OpenUpgrade.git
  target: ocb $ODOO_VERSION
  merges:
    - ocb $ODOO_VERSION
    # Example of a merge of the PR with the number <PR>
    # - oca refs/pull/<PR>/head
`

You're of great help Sir :)

**Reply by pedrobaeza** [reply]
You can fork it, or directly use GH odoo/enterprise with a proper deploy token.

**Reply by ubidev** [reply]
Where do I put the deploy token and the fork url though?
Thanks

**Reply by ap-wtioit** [reply]
Have you tried this? https://stackoverflow.com/questions/42148841/github-clone-with-oauth-access-token (make sure the urls do not end up in too many logs)

edit: using your github user and ssh seems like a better idea (without a deploy token)

**Reply by ubidev** [reply]
I'm sorry, Maybe my english is not that good after all. My understanding is that for Doodba to work once the fork is done, I have to put the fork url and key/token in Doodba. So where do I put these infos? Or am I completely off track?

**Reply by ap-wtioit** [reply, marked as answer]
in your repos.yaml you need to add the enterprise repository like this:

```yaml
./enterprise:
    defaults:
        depth: $DEPTH_DEFAULT
    remotes:
        odoo: "https://github.com/path/to/oddoo/enterprise.git"
    target:
        odoo $ODOO_VERSION
    merges:
        - odoo $ODOO_VERSION
```

and provide credentials (oauth token in the url (bad idea), .netrc (only working in case you have a user access token or only one github url) or using ssh urls and keys)

**Reply by ubidev** [reply]
Das ist großartig!
Makes alot of sense now.
Dankeschön!

**Reply by alemens** [reply]
I have no enterprise access and I need the lightest version possible of Odoo with only the Accounting and Invoicing module in a docker image. Is this possible via doodba?

**Reply by ap-wtioit** [reply]
@alemens you can limit the addons that can be installed in addons.yaml in doodba, it will however still have all the addons available in the installed branch of odoo in the image. So it technically is not the "lightest version possible" out of the box. But if you write a script that deletes all not needed addons from /opt/odoo/custom/src/odoo/addons/ and put that script in build.d to be run after [100-repos-aggregate](https://github.com/Tecnativa/doodba/blob/master/build.d/100-repos-aggregate) you can have a somewhat small image (regarding the size of odoo).

**Reply by alemens** [reply]
hi @ap-wtioit so I do not need your following suggestion?
> in your repos.yaml you need to add the enterprise repository like this:

as I said [oddoo/enterprise.git](https://github.com/path/to/oddoo/enterprise.git) respond 404 to me

**Reply by ap-wtioit** [reply]
@alemens if you don't have access to Odoo Enterprise and therefore don't know the real `/path/to/oddoo/enterprise.git` (with a typo of mine) you can only use the [account / Invoicing](https://github.com/odoo/odoo/tree/17.0/addons/account) plugin from [Odoo Community](https://www.odoo.com/page/editions) and do not need to add enterprise to your repos.yaml

If you or your customer do not want to buy Odoo Enterprise there is also a lot of companies that can help you setup Odoo with community based accounting in the [OCA](https://odoo-community.org/) one of them being [tecnativa](https://www.tecnativa.com/) the founders and maintainers of doodba.

**Reply by alemens** [reply]
@ap-wtioit the [addon / account](https://github.com/odoo/odoo/tree/17.0/addons/account) seem to be a different thing compared to [app / accounting](https://www.odoo.com/app/accounting) in fact the [editions comparison](https://www.odoo.com/page/editions) show a red flag on [accounting-features](https://www.odoo.com/app/accounting-features) for the community version 

consider what I need is just [this](https://postimg.cc/8jBn73sm) Accounting (Contabilità) module 

so I am going to loose time and mind developing on Doodba something not available for the community?

**Reply by pedrobaeza** [reply]
That thing is not depending on Doodba, but on the addons you add to it. Doodba is only a "platform", not a addon repository. You need to add the corresponding modules sources. If you don't want to pay the enterprise licenses, you have to put the corresponding OCA modules that replaces the accounting features on the enterprise modules.

I think you need to know more the ecosystem or ask in other places before blaming about the tool.

**Reply by ap-wtioit** [reply]
The "Accounting (Contabilità)" based on the icon looks very much like account_accountant from Odoo Enterprise.  (You can get the technical names for plugins in the Apps listing with using [debug features](https://www.odoo.com/documentation/17.0/applications/general/developer_mode.html)) 

![image](https://github.com/user-attachments/assets/92c6527f-a72a-449e-8f9a-cf1ac08ae537)

So you will need an enterprise license for running that on a server. [To get the source code you will most likely have to enter a partner agreement with Odoo S.A.](https://www.odoo.com/forum/help-1/enterprise-source-code-179427). Once you have the source code access you can use doodba to develop some addons for it.

**Reply by alemens** [reply]
thanks @ap-wtioit and yes @pedrobaeza [doodba](https://github.com/Tecnativa/doodba/) is not a addon repository yet I do not know if more powerful than [InstallScript](https://github.com/Yenthe666/InstallScript/)

**Reply by pedrobaeza** [reply]
That script is installing things directly on the host, while this uses Docker for isolating everything, as well as this is including a lot of tools and good practices embedded in the same system, so IMO this is more advanced, but there are opinions for all the tastes.

**Reply by ap-wtioit** [reply]
I would guess you will have the same problem with Yenthes install script: https://github.com/Yenthe666/InstallScript/issues/359

You will need access to the odoo enterprise source code first if you want to deploy it on premise.
